Task's testing & UI - resolved merge conflicts

dinhnhobao opened the pull request.

zixinn commented on src/main/java/seedu/address/logic/commands/task/TaskCommand.java:
```diff
@@ -35,8 +35,13 @@
             COMMAND_GROUP_TASK, COMMAND_WORD_SEARCH,
             PREFIX_DAY, PREFIX_MONTH, PREFIX_YEAR);
     public static final String LIST_FORMAT = String.format("%s %s", COMMAND_GROUP_TASK, COMMAND_WORD_LIST);
+    public static final String MARK_AS_DONE_FORMAT = String.format("%s %s", COMMAND_GROUP_TASK, COMMAND_WORD_DONE);
```
I think should be "%s %s %s MOD_CODE %s ID_NUMBER"

dinhnhobao replied:
Resolved. Thanks!

alushingg commented on src/main/java/seedu/address/logic/commands/task/TaskListUnDoneCommand.java:
```diff
@@ -0,0 +1,25 @@
+ */
+public class TaskListUnDoneCommand extends TaskCommand {
+
+    public static final String MESSAGE_SUCCESS = "All undone tasks in the Mod Manager are listed.";
```
Maybe this can be removed because it is not used

dinhnhobao replied:
Resolved. Thanks!

zixinn commented on docs/UserGuide.adoc:
```diff
@@ -316,11 +316,39 @@ e.g. 24:00 is not allowed.
 
 Format: `*task* list`
+==== Viewing all undone tasks
+Shows a list of all tasks across all modules in the Mod Manager. +
```
I think can rephrase this?

dinhnhobao replied:
Resolved. Thanks!

zixinn commented on docs/UserGuide.adoc:
```diff
@@ -316,11 +316,39 @@ e.g. 24:00 is not allowed.
+Shows a list of tasks for a particular module in the Mod Manager. +
+Format: `*task* module /code CS2103T`
 
 Shows a list of all tasks across all modules in the Mod Manager.
 
 Format: `*task* list`
```
This is duplicated.

dinhnhobao replied:
Resolved. Thanks!

zixinn commented on src/main/java/seedu/address/logic/commands/task/TaskCommand.java:
```diff
@@ -35,8 +35,13 @@
     public static final String LIST_FORMAT = String.format("%s %s", COMMAND_GROUP_TASK, COMMAND_WORD_LIST);
+    public static final String MARK_AS_DONE_FORMAT = String.format("%s %s", COMMAND_GROUP_TASK, COMMAND_WORD_DONE);
+    public static final String LIST_DONE_FORMAT = String.format("%s %s", COMMAND_GROUP_TASK, COMMAND_WORD_UNDONE);
+    public static final String TASK_FOR_MODULE_FORMAT = String.format("%s %s", COMMAND_GROUP_TASK, COMMAND_WORD_MODULE);
```
I think should be "%s %s %s MOD_CODE"

dinhnhobao replied:
Resolved. Thanks!

zixinn commented on docs/UserGuide.adoc:
```diff
@@ -316,11 +316,39 @@ e.g. 24:00 is not allowed.
+==== Viewing all tasks, or tasks for a specific module in ModManger
+
+Shows a list of all tasks across all modules in the Mod Manager. +
+Format: `*task* list`, `*task* list /code CS2103T`
```
I think can remove the `*task* list /code CS2103T`?

dinhnhobao replied:
Resolved! Thanks!

alushingg commented on src/main/java/seedu/address/ui/taskui/TaskListCard.java:
```diff
@@ -14,23 +14,37 @@
-    private Label id;
+    private Label moduleCode;
+    @FXML
+    private Label taskId;
```
I think it should be taskID

dinhnhobao replied:
I have to use `taskId` since `taskID` doesn't pass the `Checkstyle`

NNpanpan replied:
The class is TaskNum so ... 
Better be consistent and use taskNum?

dinhnhobao replied:
Ok let's use `taskNum` then

zixinn commented on docs/UserGuide.adoc:
```diff
@@ -316,11 +316,39 @@ e.g. 24:00 is not allowed.
+==== Deleting a task
+
+Deletes a task from the module. +
+Format: `task delete /code MOD_CODE /task TASK_NAME` +
+Example: `task delete /code CS2103T /task IP week 4`
```
I think should be task delete MOD_CODE ID_NUMBER

dinhnhobao replied:
Resolved. Thanks!

alushingg commented on src/main/java/seedu/address/logic/commands/task/TaskForOneModuleCommand.java:
```diff
@@ -0,0 +1,52 @@
+            throw new CommandException(String.format(MESSAGE_MODULE_DOES_NOT_EXIST, moduleCode));
+        }
+
+        model.updateFilteredTaskList(task -> task.getModuleCode().equals(moduleCode));
```
task.getModuleCode() returns a ModuleCode object while moduleCode is a String, so the list returned will always be empty.

dinhnhobao replied:
Resolved. Thanks!

zixinn commented on docs/UserGuide.adoc:
```diff
@@ -316,11 +316,39 @@ e.g. 24:00 is not allowed.
 ****
 
-==== Viewing all tasks
+==== Editing a task
+
+Edits the information of a task. +
+Format: `task edit MOD_CODE ID_NUMBER [/desc DESCRIPTION] [/on DATE/non] [/at TIME]` +
+Example: `task edit CS2103T 848 /desc UG submission /on 12/04/2020 /at 23:59` or `task edit CS2103T 848 /on non`
+
+==== Deleting a task
+
+Deletes a task from the module. +
+Format: `task delete /code MOD_CODE /task TASK_NAME` +
+Example: `task delete /code CS2103T /task IP week 4`
```
Editing and deleting task is already in the UG below finding upcoming task

dinhnhobao replied:
Resolved. Thanks!

dinhnhobao commented:
> For the taskForOneModule, we already have the mod view where it will show the tasks for one module, so i am wondering if it is necessary to have this command. For TaskCard.java, it uses the TaskListCard.fxml too hence mod view will not work currently since you change the TaskListCard.fxml. So if you don't mind, you can make the changes to TaskCard.java accordingly. I can do it too if you are too busy.

Hmm I don't really understand the dependency, I can change it later after midnight.

alushingg commented:
> > For the taskForOneModule, we already have the mod view where it will show the tasks for one module, so i am wondering if it is necessary to have this command. For TaskCard.java, it uses the TaskListCard.fxml too hence mod view will not work currently since you change the TaskListCard.fxml. So if you don't mind, you can make the changes to TaskCard.java accordingly. I can do it too if you are too busy.
> 
> Hmm I don't really understand the dependency, I can change it later after midnight.

It is ok I don't really know how to explain. I can make the changes after I merge yours.

dinhnhobao commented:
> * For the message usage of TaskForOneModuleCommand and TaskMarkAsDoneCommand, can add "\n" before the "Parameters:" and "Example:"?
> * For UG, can standardise whether the word task is bold or not?
> * Just curious, why you remove the argMultimap.getPreamble().isEmpty() check in the commands?
>   For example, "task delete" will hang without it.
> * Can add back the DescriptionTest?

- Thanks, added.
- Should be all bold, we should all do bold too since it looks easier to differentiate between command groups I think.
- I realised most of the commands still work with premable, so I removed them such that the code can cover more cases. I didn't realize so I just remove for TaskDelete all the way. Resolved.
- I don't remember deleting DescriptionTest. Hmm not sure where it got lost. Do you mind add it back manually?

zixinn commented:
> > * For the message usage of TaskForOneModuleCommand and TaskMarkAsDoneCommand, can add "\n" before the "Parameters:" and "Example:"?
> > * For UG, can standardise whether the word task is bold or not?
> > * Just curious, why you remove the argMultimap.getPreamble().isEmpty() check in the commands?
> >   For example, "task delete" will hang without it.
> > * Can add back the DescriptionTest?
> 
> * Thanks, added.
> * Should be all bold, we should all do bold too since it looks easier to differentiate between command groups I think.
> * I realised most of the commands still work with premable, so I removed them such that the code can cover more cases. I didn't realize so I just remove for TaskDelete all the way. Resolved.
> * I don't remember deleting DescriptionTest. Hmm not sure where it got lost. Do you mind add it back manually?

ok. we can add it back later.

dinhnhobao commented:
@zixinn @alushingg @NNpanpan Thanks for the feedback, we can merge now! Not sure the command works perfectly, I haven't tested all the Task commands yet.
